### ISSUE TITLE
GFXBIN code for editing ColorMaps results in Game Crashes

### ISSUE THREAD

**Issue by Rinual** [Kizari/Flagrum#9]
**Note for Readers**

This does not affect any of Flagrum's functionality, this is a reference to using the code to alter Vertex Colors programmatically.

Possibly direct correlation to issue #6, and the GFXBIN code for editing and removing meshes from some Models and Props with multiple meshes. 

- Zero'ing out the Ambient Occlusion Vertex Color values of character models results in crashing the game. 

- The output can then be imported into Blender with correct looking model with the AO VC all white as intended.

One instance of attempting the code did result in loading of the game, however the altered mesh were [mismatched scale from original. ](https://media.discordapp.net/attachments/997095024812621854/1003531652393357332/unknown.png?width=677&height=910) Though in continued tests, I was only able to replicate the crashes. 

```
public class AOFixer
{
    public static void Run()
    {
        var gfx = @"path\to\some\NH\model.gmdl.gfxbin";
        var gpu = gfx.Replace(".gmdl.gfxbin", ".gpubin");
        var gfxbinData = File.ReadAllBytes(gfx);
        var gpubinData = File.ReadAllBytes(gpu);
        var reader = new ModelReader(gfxbinData, gpubinData);
        var model = reader.Read();

        //change name to meshes that need targeting, i.e EyelashShape, FaceShape, BodyShape
        var meshes = new[] { "FaceShape" };
        //var meshes = new[] { "FaceShape", "BodyShape" };
        var matches = model.MeshObjects[0].Meshes.Where(m => meshes.Contains(m.Name));

        foreach (var match in matches)
        {
            foreach (var color in match.ColorMaps[2].Colors)
            {
                color.R = byte.MaxValue;
                color.G = byte.MaxValue;
                color.B = byte.MaxValue;
                color.A = byte.MaxValue;
            }
        }
        var (gfxbin, gpubin) = new ModelWriter(model).Write();
        File.WriteAllBytes(@"path\to\some\output.gmdl.gfxbin"", gfxbin);
        File.WriteAllBytes(@"path\to\some\output.gpubin", gpubin);
    }
}
```

**Comment by Kizari** [comment]
A note on the mismatched scale. We encountered this issue during the original "sin skin" fix project. Basically it would occur when trying to apply Noctis' arm material to the skin of any mods. At the time we suspected this was just a behaviour of the material itself, and perhaps something about the original Noctis model compensates for this. Perhaps whatever is broken with the GFXBIN write is also breaking whatever it is about the GFXBIN file that compensates for the scale issue?
